From 7798668160d45bbba57ddc7cf63dcb944ba0bca3 Mon Sep 17 00:00:00 2001
From: Vignesh Viswanathan <quic_viswanat@quicinc.com>
Date: Fri, 6 Oct 2023 12:18:22 +0530
Subject: [PATCH 414/500] remoteproc: qcom: q6v5: Get WCSS WDOG Crash reason
 from remote SMEM region

WCSS crash reason for WDOG crashes is also allocated by Q6 in the
SMEM remote region with id 1.
Get the WDOG crash reason from remote_id region instead of SMEM global
region.

Fixes: 768f2a3213b5 ("remoteproc: qcom: q6v5: Get WCSS Crash reason from remote SMEM region")
Change-Id: Ib1dcb33632059bf1694b0bc4faa2053ce243b75f
Signed-off-by: Vignesh Viswanathan <quic_viswanat@quicinc.com>
---
 drivers/remoteproc/qcom_q6v5.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/remoteproc/qcom_q6v5.c b/drivers/remoteproc/qcom_q6v5.c
index ee59496116cf..d42b887d6821 100644
--- a/drivers/remoteproc/qcom_q6v5.c
+++ b/drivers/remoteproc/qcom_q6v5.c
@@ -100,7 +100,7 @@ static irqreturn_t q6v5_wdog_interrupt(int irq, void *data)
 		return IRQ_HANDLED;
 	}
 
-	msg = qcom_smem_get(QCOM_SMEM_HOST_ANY, q6v5->crash_reason, &len);
+	msg = qcom_smem_get(q6v5->remote_id, q6v5->crash_reason, &len);
 	if (!IS_ERR(msg) && len > 0 && msg[0])
 		dev_err(q6v5->dev, "watchdog received: %s\n", msg);
 	else
-- 
2.34.1

